### ISSUE TITLE
[REVIEW]: ParaMonte: A high-performance serial/parallel Monte Carlo simulation library for C, C++, Fortran

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2741]
**Submitting author:** @shahmoradi (<a href="http://orcid.org/0000-0002-9720-8937">Amir Shahmoradi</a>)
**Repository:** <a href="https://github.com/cdslaborg/paramonte" target ="_blank">https://github.com/cdslaborg/paramonte</a>
**Version:** v1.5.1
**Editor:** @VivianePons
**Reviewer:** @milancurcic, @williamfgc
**Archive:** <a href="https://doi.org/10.5281/zenodo.4749957" target="_blank">10.5281/zenodo.4749957</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513/status.svg)](https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513"><img src="https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513/status.svg)](https://joss.theoj.org/papers/f964b6e22c71515c310fbe3843ad4513)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/cdslaborg/paramonte" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@milancurcic & @williamfgc, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @VivianePons know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/cdslaborg/paramonte">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@shahmoradi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [X] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [X] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/cdslaborg/paramonte">repository url</a>?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [X] **Contribution and authorship:** Has the submitting author (@shahmoradi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [X] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [X] **Installation:** Does installation proceed as outlined in the documentation?
- [X] **Functionality:** Have the functional claims of the software been confirmed?
- [X] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)  **I don't have enough domain knowledge to conduct a study on the topic**

### Documentation

- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.   Reviewer: There are build instructions, but 
- [X] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [X] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [X] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?   **There are manual steps, but not automatic CI**
- [X] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support    **Since it's hosted on GitHub it's following the typical workflow, issues, PR**

### Software paper

- [X] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [X] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [X] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @milancurcic, @williamfgc it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1088/0004-637x/766/2/111 may be a valid DOI for title: A Multivariate Fit Luminosity Function and World Model for Long Gamma-Ray Bursts
- 10.1093/mnras/stv714 may be a valid DOI for title: Short versus long gamma-ray bursts: a comprehensive study of energetics and prompt gamma-ray correlations
- 10.1007/s11222-006-9438-0 may be a valid DOI for title: DRAM: efficient adaptive MCMC
- 10.1093/bioinformatics/btp162 may be a valid DOI for title: GNU MCSim: Bayesian statistical inference for SBML-coded systems biology models
- 10.18637/jss.v035.i04 may be a valid DOI for title: PyMC: Bayesian stochastic modelling in Python

INVALID DOIs

- None
```

**Comment by williamfgc** [comment]
@whedon @VivianePons I have trouble when trying to accept the invitation above. Also, I can't edit my checklist. I'd appreciate help with it as I'm currently reviewing the software.

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by williamfgc** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by williamfgc** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by VivianePons** [comment]
@whedon re-invite @williamfgc as reviewer

**Comment by VivianePons** [comment]
That should fix the issue

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@williamfgc please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by williamfgc** [comment]
I have a few initial recommendations upon my review. I hope the authors find this helpful to transition ParaMonte from academic code to production quality software:

1. I couldn't find the use of automated test frameworks for unit and functional tests. I'd recommend using `ctest` as the project has CMake support and create automated tests. It seems there are a set of tests under [src/test](https://github.com/cdslaborg/paramonte/blob/master/src/test/CMakeLists.txt), but they are not executed. Also a [Fortran unit testing framework](http://fortranwiki.org/fortran/show/Unit+testing+frameworks) could be a good alternative.
2. CI: this project is missing continuous integration. I find it difficult to accept it without comprehensive testing when changes are made to the repo. CI builds trusts among users.
3. Contributing: it's not clear the contribution policy. This is tied to 3. and the lack of CI infrastructure (either GitHub Actions, CircleCI, Travis, etc.). 
4. Installation: add support for CMake installation. While there is a large effort to support different platform via scripts, it would be beneficial to reuse CMake portable capabilities to build and install the library portably. This could be added to their CI.
5. Code coverage: once CI is installed please add code coverage (e.g. Coverity, Codecov) 
6. Code documentation: it's hard to find API level documentation (e.g. doxygen) so any person new to the project can pick up the author's intention.
7. The paper focuses too much on previous work publications, rather than exposing the merits of the software infrastructure and (most importantly) impact. I'd focus it on the community impact: use cases, important integration with other projects, etc.
8.  Distribution and packaging (optional): it would be good to have the option for standard distribution packages (conda, spack). Install scripts have to be constantly updated and maintained. Without proper CI, bugs could pile up pretty quickly.

I'd be happy to expand on the above if required, but I believe they are necessary to meet minimal software quality standards.

**Comment by VivianePons** [comment]
Hi @shahmoradi have you had any time to look at the comments made by @williamfgc ?

**Comment by shahmoradi** [comment]
@VivianePonsThank you for your reminder. I am working on it right now as we speak.

**Comment by milancurcic** [comment]
The items I checked off are okay. Few remain unchecked because of issues with installation.

I tested serial binary releases for Fortran with both GNU and Intel. Both worked as expected and I was able to build and run the example program.

I wasn't able to use the parallel releases because they require MPICH. I tend to work with OpenMPI so it'd be nice if binary releases with OpenMPI were available. I opened an [issue](https://github.com/cdslaborg/paramonte/issues/5) for this.

I tried building from source which failed. I opened an [issue](https://github.com/cdslaborg/paramonte/issues/3) for this.

There was also a slight issue with wording in [ACKNOWLEDGMENT.md](https://github.com/cdslaborg/paramonte/blob/master/ACKNOWLEDGMENT.md) which make it sound like the license requires citing the software. I opened an [issue](https://github.com/cdslaborg/paramonte/issues/4) for this.

I confirm and re-iterate points 1 and 3 by @williamfgc as important. Points 2, 5, and 8 I consider nice to have but not essential. Point 6 I couldn't confirm--I had no problem finding the API docs and found them thorough.

I suggest tackling these before proceeding with the review.

**Comment by williamfgc** [comment]
@milancurcic thanks for confirming. I modified some points to make them clear based on your comments.

**Comment by VivianePons** [comment]
Hi all, @shahmoradi can I ask where we are on this? (No rush, I just want to make sure we are still on track!)

**Comment by shahmoradi** [comment]
I have addressed @williamfgc comments and I am working on @milancurcic 's comments. I should be able to back with a full revision by tomorrow, hopefully.

**Comment by shahmoradi** [comment]
@williamfgc

Thank you for your extensive detailed feedback, which significantly improved the quality of this codebase. We will address each of your comments below.  

1.  **The library tests**.  
    The library originally had a significant number of tests that used to be automatically executed. But at some point, someone (perhaps myself) decided to temporarily disable the tests, which then became permanent since nobody reactivated the tests. 

    We have now added a comprehensive set of 866 new unique tests that cover nearly 100% of the entire library as well as the various functionalities of the samplers in the library. Given the three possible major parallelism configurations of the library, there are overall `3 * 866 = 2598` tests for all serial as well as the MPI and Coarray parallel builds. The code coverage report for all three builds for the most recent version of the library is available here:
    + [serial](https://cdslaborg.github.io/paramonte-codecov/kernel/serial/)  
    + [MPI parallel](https://cdslaborg.github.io/paramonte-codecov/kernel/mpi/)  
    + [Coarray parallel](https://cdslaborg.github.io/paramonte-codecov/kernel/caf/)  

    Since the library relies on preprocessing directives to implement different parallelism paradigms (serial, MPI, and Coarray Fortran), the tests had to be also adapted and preprocessed according to the characteristics of the library for different parallelism paradigms. No testing framework that we have inspected so far, met the minimum requirements for testing this package under the different build configurations. We have therefore developed a unit testing framework that is specifically tailored for the needs of this library.
    The tests in the library are separated into two different categories:  
    1.  `basic`: Tests that verify the functionality and accuracy of the fundamental backbone modules and procedures of the library.  
    1.  `sampler`: Tests that verify the functionality, semantics, and accuracy of the library's samplers.  

    By default both categories of tests are deactivated for any particular build for the following reasons:
    1.  The tests can take significant time and memory, in particular, when building the library for multiple different configurations. This can also easily break Continuous Integration workflows like Travis CI that limit the maximum length of the stdout log file.  
    2.  Some of the tests such as those that verify the restart functionality of the library's sampler in parallel require external intervention in the regular working of the samplers. This in turn requires the MPI/Coarray processes to be able to communicate the error messages with each other to avoid deadlocks. These error message communications between different processes are not normally needed as the failing process automatically calls `mpi_abort()` in MPI or `error stop` in Coarray to halt the entire program.  
    However, such exceptions and errors should be gracefully handled during testing, which requires the addition of extra error handling code snippets to the sampler routines that are otherwise deactivated in normal production runs.  
    These extra exception and error handling communications in parallel modes can be quite expensive if they appear in the parallel production code. As such, the library's tests are by default switched off, unless the developer (or the user) readily but explicitly enables the tests via the `-t all` or `--test all` or `--codecov` build flag. For example, 
    ```bash  
    ./install.sh --lang fortran --build testing --lib dynamic --par none --test all
    ```  
    will build the library for usage from the Fortran language with all tests activated.  The tests can be also limited to one category, if desired, to reduce the testing and build time. This is particularly needed during the development of the library when the focus is on one particular functionality in the library. For example,  
    ```bash  
    ./install.sh --lang c --build testing --lib dynamic --par mpi -t basic
    ```  
    will test the library build for parallel usage from the C language and will only activate the basic tests of the library. The default mode, as mentioned above, is,  
    ```bash  
    ./install.sh -t none
    ```  
    However, when generating code coverage, all tests are activated by default. For example,  
    ```bash  
    ./install.sh --codecov
    ```  
    and,  
    ```bash  
    ./install.sh --codecov -t all
    ```  
    are equivalent and,  
    ```bash  
    ./install.sh --codecov -t none
    ```  
    leads to an error message, since a minimal set of tests are required to generate the code coverage report.  

2.  **Continuous Integration**.  
    We have now enabled [Travis CI](https://travis-ci.org/github/cdslaborg/paramonte/builds/749668477) for this library. All builds are successful. Due to technical difficulties with Coarray Fortran and Windows, CAF parallel builds and builds for Windows are currently excluded from Travis CI. Intel has recently released its new version of Fortran compilers free of charge for all platforms including Windows. This will hopefully enable the testing and CI of this library also on Windows in the near future.  

3.  **Contribution policy**.  
    We have now laid out a set of guidelines for contributing to the core of library or other contributions:  
    +   [Defect reports](https://www.cdslab.org/paramonte/notes/contributing/defects/)  
    +   [Enhancement requests](https://www.cdslab.org/paramonte/notes/contributing/enhancements/)  
    +   [Development - general guidelines](https://www.cdslab.org/paramonte/notes/contributing/development/general/)
    +   [Development - core library guidelines](https://www.cdslab.org/paramonte/notes/contributing/development/kernel/)

4. **CMake support**.  
    CMake is indeed already supported on Linux and macOS. However, direct use of CMake to build the library is suboptimal, in particular, for the end-users as CMake cannot install missing libraries on the user's system if needed. One of our primary goals in the development of this package was to make a compiled-language library as accessible as possible to anyone, even those who do not have experience with compiled languages or CMake or different library dependencies. To achieve this goal, we had to develop multiple layers of build scripts that lay the ground for CMake to successfully build the library. The current build scripts are capable of automatically installing all missing components including the CMake application (if the minimum required version is not detected on the system), GNU compilers, as well as OpenMPI, MPICH MPI, and OpenCoarrays libraries.   

    The situation on Windows is slightly more complex. While library build with CMake was originally supported on Windows, we eventually dropped it in favor of developing a home-grown separate Windows Batch build system for the library exclusively on Windows. The reason for this decision was CMake's poor support of the Fortran programming language at the time, especially on Windows. The CMake has dramatically improved over the past three years and given the recent availability of Intel compilers on Windows free of charge, CMake support for Windows is now among top priorities to add to the library.  

5.  **Code coverage**.  
    Code coverage reports are now added for every new release of the library and are permanently available on GitHub in a separate repository for the three most different builds of the library (the three parallelism paradigms):
    + [serial](https://cdslaborg.github.io/paramonte-codecov/kernel/serial/)  
    + [MPI parallel](https://cdslaborg.github.io/paramonte-codecov/kernel/mpi/)  
    + [Coarray parallel](https://cdslaborg.github.io/paramonte-codecov/kernel/caf/)  

    We have strived for adding an automated code coverage report via [codecov](https://codecov.io/gh/cdslaborg/paramonte) but so far in vain. We will continue our efforts to generate automated code coverage reports and analyses via codecov/TravisCI or similar platforms.  

6.  **Documentation**.  
    + The library documentation for the end-users is available here: https://www.cdslab.org/paramonte/index.html  
    + The library's core API documentation is also available through the same link in the above, or directly at: https://cdslaborg.github.io/paramonte-kernel-doc/html/

7.  **Manuscript's contents and focus**.  
    We have now significantly reduced the amount of technical contents of the paper.  

8.  **Package redistribution and CI**.  
    We absolutely agree that CI is necessary for the success and longevity of this repository and we have now set up an automated build and testing workflow with Travis-CI. We also agree that having a standard distribution method is essential in the long term. There have been, however, a few challenges along our way on identifying and setting up the best method of distribution, most importantly, the many different configurations with which the library can be built, such as the programming languages, the MPI library dependencies, etc. Furthermore, there is the question of which packaging and redistribution method would be the best. On Windows and macOS, the task might be rather straightforward as only a few popular methods exist. The situation, however, is completely different on Linux OS.  

    Currently, the users can access the prebuilt versions of the library for almost any production configuration they might need. The prebuilt libraries are always available on [the GitHub release page of the project](https://github.com/cdslaborg/paramonte/releases). Downloading and using these prebuilt libraries takes only three steps from the bash terminal. For example, the prebuilt library for the C programming language can be downloaded and run via,  
    ```bash  
    libname=libparamonte_c_linux_x64_intel_release_dynamic_heap
    wget https://github.com/cdslaborg/paramonte/releases/latest/download/$libname.tar.gz
    tar xvzf $libname.tar.gz && cd $libname && ./build.sh && ./run.sh && cd ..
    ```  

@milancurcic

Thank you for your valuable feedback, which significantly improved the quality of this work. Below, we will address your comments and questions,  

1.  **OpenMPI releases**.  
    Thank you for pointing this out. As you mentioned, OpenMPI is a popular implementation. We have now added the prebuilt versions of the library with OpenMPI in addition to MPICH. This required a complete revision of the naming convention used in the library across all languages and documentations. The new naming convention automatically suffixes,  
    1.  any MPICH build of the library with *_mpich*,  
    1.  any OpenMPI build of the library with *_openmpi*,  
    1.  any Intel MPI build of the library with *_impi*,  
    A complete description of the naming convention used in the library is provided in [the library's documentation](https://www.cdslab.org/paramonte/notes/installation/readme/).  

2.  **Build failure**.  
    Thank you for pointing this out. This apparently happens on systems where the either the MPI library or the GNU compilers are already installed and meet the minimum version requirements of ParaMonte. As such, the build scripts would not feel the need to create a custom `setup.sh` to set up the environmental variables. However, it would require the setup file's existence, which would subsequently interrupt the build by throwing an exception.  

    We believe this issue is now fixed in [the latest release of the library](https://github.com/cdslaborg/paramonte/archive/main.tar.gz).  

3.  **license and ACKNOWLEDGMENT.md**.  
    Thank you again for pointing this out. The existing `ACKNOWLEDGMENT.md` was a relic of the original license of the library. [This is now fixed](https://github.com/cdslaborg/paramonte#citing-paramonte).  

4.  **points 1 and 3 by @williamfgc**.  
    Please see our responses to comments by @williamfgc.

**Comment by shahmoradi** [comment]
@milancurcic @williamfgc 
Thank you again for your constructive feedback so far. Just to make sure we have not missed any of your comments, are there any other outstanding issues needed to address here?

**Comment by williamfgc** [comment]
Dear @shahmoradi , thanks for the hard work on getting this points addressed. Would you kindly send an updated version of your manuscript? I'd like to wrap up the review as well.

**Comment by shahmoradi** [comment]
@williamfgc Thank you. My impression was that @whedon would automatically build the PDF from the [Markdown file](https://github.com/cdslaborg/paramonte/blob/main/pubs/ParaMonte/kernel/JOSS2020/paper.md). Until then, I attach a PDF of the revised draft to this comment. The Markdown file on GitHub is also readable, although not as comfortable as the PDF.

**Comment by VivianePons** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by williamfgc** [comment]
@shahmoradi I just reviewed the revised manuscript and looks appropriate for JOSS. Thanks for addressing the changes and fixing the defect I filed above. Minor thing: Funding and Acknowledgment can be one single section on the paper. Looks good to me, the only one thing I still think could be added in the future is CI, but I'll leave it to @VivianePons as the editor and @milancurcic to decide on final publication. Thanks!

**Comment by milancurcic** [comment]
Good to go from me, thank you all. :+1: 

A note regarding "Functionality" and why I left it unchecked: I was reading through the docs and played for a while with the example program and configurations, however I wasn't able to get far, even with the docs. I believe this was due to the domain-specific knowledge that I don't have. But I assume that for a domain expert, these parameters are meaningful and they wouldn't have trouble generalizing the example application while following the docs.

**Comment by VivianePons** [comment]
Hi thank you both @williamfgc and @milancurcic for the thorough reviews and @shahmoradi for all the improvements.

@williamfgc you left many boxes unchecked even though I understand that you are satisfied with the software. Maybe that's just an oversight? Could you go through the list and see if everything checks (and indeed CI is not required even though it's always good to have)

@milancurcic is there any functionality that you were not able to reproduce? Or anything you wanted to try and were not able to do? Even if you feel you're not a domain expert, your input might be useful for proving the documentation for example (and as an editor, I would really much like seeing this "functionality" box checked to get the paper published!) Also, you left unchecked the "Performance" box (even there is no performance claims in the paper, it should be checked)

Again thank you for your time! I believe we are getting close to final publication!

**Comment by williamfgc** [comment]
@VivianePons yes, it's just an oversight from my side. I just went through the list and marked what is completed. Thanks for double checking and leading the review process.

**Comment by milancurcic** [comment]
@VivianePons 

> Also, you left unchecked the "Performance" box (even there is no performance claims in the paper, it should be checked)

Sorry, I misunderstood the prompt. Checked!

> @milancurcic is there any functionality that you were not able to reproduce? Or anything you wanted to try and were not able to do?

No, sorry, I wasn't clear. Everything I tried *seems* to work and do things. What I struggled with was fully understanding what it all means, which is the domain-specific part. I will check the "Functionality" box to the best of my knowledge, however, I believe it will take a domain-expert to truly confirm that the software does what it says. I know and trust the Authors so I feel comfortable checking this box.

**Comment by VivianePons** [comment]
Thanks @milancurcic for the clarification, I will keep it in mind.

@williamfgc please let us know if you see extra changes to be made to the software / paper or if you have any suggestions while finishing the review

**Comment by williamfgc** [comment]
@VivianePons it's all good from my side, the points in my original review were addressed. Thanks!

**Comment by VivianePons** [comment]
Thank you very much @williamfgc ! I see that you have left unchecked the Performance box (it need to be checked if there are no performance claims in the paper) and also "automated tests" and "Community guidelines", is there anything to be done still?

Also, @milancurcic was not completely confident checking the Functionality box mostly because he was not a domain expert and did not fully understand the functionality of the software. Can you tell me to what extent you were able to follow on your end?

**Comment by williamfgc** [comment]
@VivianePons I checked the boxes and put some follow up comments in bold. I don't want to be the bottleneck for final publication :) . Thanks for guiding the JOSS review process, I enjoyed it.

**Comment by VivianePons** [comment]
Thank you very much! And thanks again @milancurcic These have been very thorough review and I believe the quality of the software has greatly increased

I will now start the final steps towards publication

**Comment by VivianePons** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK

MISSING DOIs

- 10.1088/0004-637x/766/2/111 may be a valid DOI for title: A Multivariate Fit Luminosity Function and World Model for Long Gamma-Ray Bursts
- 10.1093/mnras/stv714 may be a valid DOI for title: Short versus long gamma-ray bursts: a comprehensive study of energetics and prompt gamma-ray correlations
- 10.3847/1538-4357/abb9b7 may be a valid DOI for title: A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts
- 10.3847/1538-4357/abb9b7 may be a valid DOI for title: A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts
- 10.1007/s11222-006-9438-0 may be a valid DOI for title: DRAM: efficient adaptive MCMC
- 10.1093/bioinformatics/btp162 may be a valid DOI for title: GNU MCSim: Bayesian statistical inference for SBML-coded systems biology models
- 10.18637/jss.v035.i04 may be a valid DOI for title: PyMC: Bayesian stochastic modelling in Python

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
Hi @shahmoradi 

I believe you're missing a word in this sentence "the library was as an open-source project" line 72 of the PDF.

Also, could you check the missing DOI that were flagged by whedon?

Thank you

**Comment by shahmoradi** [comment]
> 
> 
> Good to go from me, thank you all. 👍
> 
> A note regarding "Functionality" and why I left it unchecked: I was reading through the docs and played for a while with the example program and configurations, however I wasn't able to get far, even with the docs. I believe this was due to the domain-specific knowledge that I don't have. But I assume that for a domain expert, these parameters are meaningful and they wouldn't have trouble generalizing the example application while following the docs.

@milancurcic Thanks for pointing this out. Such feedbacks are highly valuable, Could you tell me which examples you tried, so that we review them again for accuracy and comprehensibility? Thanks again.

**Comment by shahmoradi** [comment]
> 
> 
> @shahmoradi I just reviewed the revised manuscript and looks appropriate for JOSS. Thanks for addressing the changes and fixing the defect I filed above. Minor thing: Funding and Acknowledgment can be one single section on the paper. Looks good to me, the only one thing I still think could be added in the future is CI, but I'll leave it to @VivianePons as the editor and @milancurcic to decide on final publication. Thanks!

Thank you @williamfgc . Your comments along with @milancurcic 's were highly valuable and crucial to making this a quality software, although there is still ample room for perfection. I will include your minor revision requests in the final draft.

**Comment by shahmoradi** [comment]
@VivianePons Thank you for your corrections. I confirm that the DOIs are correct to the best of my knowledge. I have also revised the manuscript based on the latest corrections requested by @VivianePons and @williamfgc. If there is anything else to implement, please let me know. Thank you again for your excellent edit and reviews.

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK

MISSING DOIs

- 10.1088/0004-637x/766/2/111 may be a valid DOI for title: A Multivariate Fit Luminosity Function and World Model for Long Gamma-Ray Bursts
- 10.1093/mnras/stv714 may be a valid DOI for title: Short versus long gamma-ray bursts: a comprehensive study of energetics and prompt gamma-ray correlations
- 10.3847/1538-4357/abb9b7 may be a valid DOI for title: A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts
- 10.3847/1538-4357/abb9b7 may be a valid DOI for title: A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts
- 10.1007/s11222-006-9438-0 may be a valid DOI for title: DRAM: efficient adaptive MCMC
- 10.1093/bioinformatics/btp162 may be a valid DOI for title: GNU MCSim: Bayesian statistical inference for SBML-coded systems biology models
- 10.18637/jss.v035.i04 may be a valid DOI for title: PyMC: Bayesian stochastic modelling in Python

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
👋 @shahmoradi  - this is either waiting for your action, or for you to tell the editor that you have completed it.

**Comment by kyleniemeyer** [comment]
Hello @shahmoradi, just checking in again. We are essentially ready to accept your submission, but just need you to make that change to your article.

**Comment by shahmoradi** [comment]
My Apologies for the delay, I have become fully occupied with some personal
and family matters over the past month. Please give me until this weekend
to fix the outstanding issues and resubmit this paper to JOSS. Thank you.

**Comment by kyleniemeyer** [comment]
@shahmoradi no problem, just let us know

**Comment by shahmoradi** [comment]
@whedon generate pdf

**Comment by shahmoradi** [comment]
Thank you for your patience and I apologize again for the delay. All DOIs should be now fixed. If there is anything else I have missed, please let me know.
Best regards,

**Comment by whedon** [comment]
PDF failed to compile for issue #2741 with the following error: 
```
 Error reading bibliography file paper.bib:
(line 245, column 3):
unexpected "p"
expecting space, ",", white space or "}"
Looks like we failed to compile the PDF

```

**Comment by shahmoradi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK
- 10.1088/0004-637x/766/2/111 is OK
- 10.1093/mnras/stv714 is OK
- 10.3847/1538-4357/abb9b7 is OK
- 10.1007/s11222-006-9438-0 is OK
- 10.1093/bioinformatics/btp162 is OK
- 10.18637/jss.v035.i04 is OK

MISSING DOIs

- 10.3847/1538-4357/abb9b7 may be a valid DOI for title: A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
@wheon generate pdf

**Comment by VivianePons** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by VivianePons** [comment]
It seems that you have the same paper appearing twice in the bib file (A Multilevel Empirical Bayesian Approach to Estimating the Unknown Redshifts of 1366 BATSE Catalog Long-Duration Gamma-Ray Bursts)

**Comment by shahmoradi** [comment]
Fixed.

**Comment by shahmoradi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2741 with the following error: 
```
 Error reading bibliography file paper.bib:
(line 146, column 1):
unexpected "%"
expecting space, white space or "}"
Looks like we failed to compile the PDF

```

**Comment by shahmoradi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK
- 10.1088/0004-637x/766/2/111 is OK
- 10.1093/mnras/stv714 is OK
- 10.1007/s11222-006-9438-0 is OK
- 10.1093/bioinformatics/btp162 is OK
- 10.18637/jss.v035.i04 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
Thank you @shahmoradi for the fixes! Could you make an archive (on Zenodo, figshare, or other) of the software and give me the DOI? Be careful to use the same metadata as the paper (title and authors)

Thank you

**Comment by shahmoradi** [comment]
Here is the DOI (https://zenodo.org/record/4739757#.YJNelqFOnGp):
10.5281/zenodo.4739757
Sincerely,

**Comment by shahmoradi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02741/joss.02741/10.21105.joss.02741.pdf) :page_facing_up: :point_left:

**Comment by VivianePons** [comment]
Hi @shahmoradi it looks like you have archived the wrong thing. I see only the pdf paper and not the whole software repo.

**Comment by shahmoradi** [comment]
My apologies. I am not quite familiar with zenodo and other DOI granting websites. I have uploaded the repository to Zenodo. Does this look good? https://zenodo.org/record/4739775#.YJRg26FOnGo

**Comment by VivianePons** [comment]
Well, it's a bit weird because now you have the whole thing as a zip. The archive should be a direct copy of the repo, like you did in previous versions for example here: https://zenodo.org/record/4414285#.YJTwMiY69H4

**Comment by shahmoradi** [comment]
Could we then use the DOI you mentioned? This is DOI that I generated for the project last year. The new one that is troubling is the one that I tried to generate yesterday. If none work, I will retry fresh. Thank you for your patience.

**Comment by VivianePons** [comment]
We need a DOI with the current version of the software, including all changes made during the review so we can't use the old one

**Comment by shahmoradi** [comment]
@VivianePons   I think the current zenodo DOI is automatically connected to the latest version of the software. See for example the history of the version of the package on this page: https://zenodo.org/record/4739775#.YJXr9aFOnGr
I can remove the PDF of the paper that I added manually to the zenodo entry yesterday. If this does not help either, then I will create a DOI via figshare or some other platform as I am somewhat confused now on how Zenodo works and how to create new fresh entries of the same repository.
Thank yo

**Comment by VivianePons** [comment]
Hi @shahmoradi each DOI is linked to a specific version, I need the current version. The archive is like a picture of the current state of the software, which I need to finalize the publication, so that, in the future it is clear that the paper refers to this specific state of the software. Please either add a proper archive on the current zenodo or make a new one elsewhere.

**Comment by shahmoradi** [comment]
Here is the new Zenodo entry: https://zenodo.org/record/4749957#.YJrS7qFOnGo
DOI: 10.5281/zenodo.4749957

Thank you
Amir

**Comment by VivianePons** [comment]
@whedon set 10.5281/zenodo.4749957 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4749957" target="_blank">10.5281/zenodo.4749957</a> is the archive.

**Comment by VivianePons** [comment]
@whedon set v1.5.1 as version

**Comment by whedon** [comment]
OK. v1.5.1 is the version.

**Comment by VivianePons** [comment]
Thank you very much @shahmoradi we are now ready to accept the paper! Thanks to both reviewers @milancurcic and @williamfgc for their work

**Comment by VivianePons** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK
- 10.1088/0004-637x/766/2/111 is OK
- 10.1093/mnras/stv714 is OK
- 10.1007/s11222-006-9438-0 is OK
- 10.1093/bioinformatics/btp162 is OK
- 10.18637/jss.v035.i04 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2301

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2301, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @shahmoradi - I'm the AEiC on duty currently, and I'll be proofreading this shortly, then either requesting changes, or proceeding to publication.

**Comment by danielskatz** [comment]
My suggested changes are in https://github.com/cdslaborg/paramonte/pull/14 - please merge this, or let me know what you disagree with, then we can proceed

**Comment by shahmoradi** [comment]
done. Thank you @danielskatz @VivianePons for your patience and edits, and special thanks again to both reviewers @milancurcic and @williamfgc for their variable suggestions.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2303

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2303, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3847/1538-4357/abb9b7 is OK
- 10.1088/0004-637x/766/2/111 is OK
- 10.1093/mnras/stv714 is OK
- 10.1007/s11222-006-9438-0 is OK
- 10.1093/bioinformatics/btp162 is OK
- 10.18637/jss.v035.i04 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by danielskatz** [comment]
👋 @VivianePons - this paper is probably too long and too detailed, but I don't think it's worth cutting it at this point - this is just a thought of something to consider for other future papers

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1392833044705333251) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2304
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02741](https://doi.org/10.21105/joss.02741)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @shahmoradi (Amir Shahmoradi) and co-author!!

And thanks to @milancurcic and @williamfgc for reviewing, and @VivianePons for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02741/status.svg)](https://doi.org/10.21105/joss.02741)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02741">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02741/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02741/status.svg
   :target: https://doi.org/10.21105/joss.02741
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02741/status.svg)](https://doi.org/10.21105/joss.02741)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by danielskatz** [comment]
@arfon and @shahmoradi - I just realized (due to a comment by @kyleniemeyer in another review) that I made a mistake when suggesting edits to this paper, which removed some of the references.  Specifically, in https://github.com/cdslaborg/paramonte/pull/14, I put multiple references together with commas, instead of the correct semicolons, which led to all but the first of each set not being included in the pdf.

I'm quite sorry about this.

I'm not sure how to fix this, as I think the paper src has been deleted.

Can we fix this?  If so, how?

**Comment by shahmoradi** [comment]
Thanks for the note. I have revived the manuscript Markdown file in the project's repository. Please let me know if I have to do anything else. Thank you.

**Comment by danielskatz** [comment]
Thanks @shahmoradi - can you merge https://github.com/cdslaborg/paramonte/pull/15

**Comment by arfon** [comment]
> Can we fix this? If so, how?

I think this should be fixed in https://github.com/openjournals/joss-papers/commit/c529ab42dd63f73bccc42b105115deca72b468c8

**Comment by danielskatz** [comment]
The two references at the end of the paragraph that starts with "For each parallel simulation" also should be put together, I think

**Comment by danielskatz** [comment]
And the same with the two at the end of the paragraph that starts with "To alleviate"

**Comment by arfon** [comment]
Done and done. The updated paper might take a few hours to show up due to caching but it should be fixed.

**Comment by danielskatz** [comment]
Thanks! And again, sorry for this mistake on my part.
